infra/build/functions/build_lib.py

```diff
@@ -410,13 +414,14 @@ def get_docker_build_step(image_names,
   if cache_image:
     args.extend(['--build-arg', f'CACHE_IMAGE={cache_image}'])
 
-  for image_name in image_names:
+  for image_name in sorted(image_names):
     args.extend(['--tag', image_name])
 
   step = {
       'name': DOCKER_TOOL_IMAGE,
       'args': args,
       'dir': directory,
+      'id': f'build-{get_unique_build_step_image_id()}',
```

Comment:
can you add a comment to explain what this is needed for?

Reply:
This awful bit of code (awful because the tests make it awful, because I need to change 5 or 6 3000 line golden files) is needed to allow the push step to happen concurrently with compilation.